pkg/plugins/runtime/gateway/generator.go

```diff
@@ -125,10 +119,7 @@ func (g Generator) Generate(ctx xds_context.Context, proxy *core_xds.Proxy) (*co
 	resources := ResourceAggregator{core_xds.NewResourceSet()}
 
 	// Cache external services since multiple listeners might need them.
-	externalServices, err := listResources(manager, core_mesh.ExternalServiceType)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to list ExternalServices")
-	}
+	externalServices := ctx.Mesh.Resources.ExternalServices()
```

Comment:
you can inline this when `GatewayResourceInfo` is created in 152. We don't need to cache it, `ctx.Mesh.Resources` is a cache on its own

Reply:
I'll remove the comment about caching, as it's no longer really a cache. But I'd still rather just call the function once outside the loop since the variable doesn't change.